networkx/generators/classic.py

```diff
@@ -617,21 +672,32 @@ def complete_multipartite_graph(*block_sizes):
 
     See also
     --------
-
     complete_bipartite_graph
-
     """
-    G = nx.empty_graph(sum(block_sizes))
-    # If block_sizes is (n1, n2, n3, ...), create pairs of the form (0, n1),
-    # (n1, n1 + n2), (n1 + n2, n1 + n2 + n3), etc.
-    extents = zip([0] + list(accumulate(block_sizes)), accumulate(block_sizes))
-    blocks = [range(start, end) for start, end in extents]
-    for (i, block) in enumerate(blocks):
-        G.add_nodes_from(block, block=i)
-    # Across blocks, all vertices should be adjacent. We can use
-    # itertools.combinations() because the complete multipartite graph is an
-    # undirected graph.
+    # The complete multipartite graph is an undirected simple graph.
+    G = nx.Graph()
+    G.name = 'complete_multiparite_graph{}'.format(block_sizes)
+
+    if len(block_sizes) == 0:
+        return G
+
+    # set up blocks of nodes
+    try:
+        extents = pairwise(accumulate((0,) + block_sizes))
+        blocks = [range(start, end) for start, end in extents]
+    except TypeError:
```

Comment:
Just to be clear, the TypeError occurs because of accumulate, but only when the iterator has to __add__ -- which doesn't happen until the list comprehension starts the iterators. So the ```extents =``` line could theoretically be moved outside the ```try```.